File: easy/readMore/readMore.cs
using System;
using System.IO;
using System.Collections.Generic;
class program
{
    static void Main(string[] args)
    {
        //using (StreamReader sr = new StreamReader("test_file.txt"))
        using (StreamReader sr = File.OpenText(args[0]))
        {
            string line, output, sub;

            while ((line = sr.ReadLine()) != null)
            {

                sub = (line.Length <= 55) ? line : line.Substring(0, 40);
                char[] arr = sub.ToCharArray();
                Array.Reverse(arr);
                string reverse = new string(arr);

                var spaceIndex = reverse.IndexOf(" ");
               
                output = (line.Length <= 55) ? line : line.Substring(0, 39 - spaceIndex) + "... <Read More>";
                                   
                Console.WriteLine(output);
            }
        }
        //Console.Read();
    }
}

File: easy/readMore/readMore.rb
File.open(ARGV[0]).each_line do |line|
	line = line.chomp
	if line.length <= 55
		puts line
	elsif line.length > 0
		sub = line[0..39].reverse
		#puts sub
		#puts sub.index(' ')
		spaceIndex = sub.index(' ') == nil ? 0 : sub.index(' ') + 1
		sub = sub.reverse
		puts sub[0..39-spaceIndex] + "... <Read More>"
	end
end
